src/Effect/ReadFS.hs

```diff
@@ -215,81 +275,91 @@ type Parser = Parsec Void Text
-readContentsYaml file = context ("Parsing YAML file '" <> toText (fromAbsFile file) <> "'") $ do
+readContentsYaml file = context ("Parsing YAML file '" <> toText (toString file) <> "'") $ do
   contents <- readContentsBS file
   case decodeEither' contents of
-    Left err -> fatal $ FileParseError (fromAbsFile file) (toText $ prettyPrintParseException err)
+    Left err -> fatal $ FileParseError (toString file) (toText $ prettyPrintParseException err)
     Right a -> pure a
 
 -- | Read XML from a file
 readContentsXML :: (FromXML a, Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m a
-readContentsXML file = context ("Parsing XML file '" <> toText (fromAbsFile file) <> "'") $ do
+readContentsXML file = context ("Parsing XML file '" <> toText (toString file) <> "'") $ do
   contents <- readContentsText file
   case parseXML contents of
-    Left err -> fatal $ FileParseError (fromAbsFile file) (xmlErrorPretty err)
+    Left err -> fatal $ FileParseError (toString file) (xmlErrorPretty err)
     Right a -> pure a
 
 type ReadFSIOC = SimpleC ReadFSF
 
 runReadFSIO :: Has (Lift IO) sig m => ReadFSIOC m a -> m a
 runReadFSIO = interpret $ \case
   ReadContentsBS' file -> do
-    BS.readFile (fromSomeFile file)
-      `catchingIO` FileReadError (fromSomeFile file)
+    BS.readFile (toString file)
+      `catchingIO` FileReadError (toString file)
   ReadContentsBSLimit' file limit -> do
     readContentsBSLimit' file limit
-      `catchingIO` FileReadError (fromSomeFile file)
+      `catchingIO` FileReadError (toString file)
   ReadContentsText' file -> do
-    (decodeUtf8 <$> BS.readFile (fromSomeFile file))
-      `catchingIO` FileReadError (fromSomeFile file)
+    (decodeUtf8 <$> BS.readFile (toString file))
+      `catchingIO` FileReadError (toString file)
   ResolveFile' dir path -> do
     PIO.resolveFile dir (toString path)
-      `catchingIO` ResolveError (toFilePath dir) (toString path)
+      `catchingIO` ResolveError (toString dir) (toString path)
   ResolveDir' dir path -> do
     PIO.resolveDir dir (toString path)
-      `catchingIO` ResolveError (toFilePath dir) (toString path)
+      `catchingIO` ResolveError (toString dir) (toString path)
   ListDir dir -> do
     PIO.listDir dir
-      `catchingIO` ListDirError (toFilePath dir)
+      `catchingIO` ListDirError (toString dir)
   GetIdentifier dir -> do
-    (extractIdentifier <$> Posix.getFileStatus (toFilePath dir))
-      `catchingIO` FileReadError (toFilePath dir)
+    (extractIdentifier <$> Posix.getFileStatus (toString dir))
+      `catchingIO` FileReadError (toString dir)
     where
       extractIdentifier :: Posix.FileStatus -> DirID
       extractIdentifier status =
         let (CIno fileID) = Posix.fileID status
             (CDev devID) = Posix.deviceID status
          in DirID{dirFileID = toInteger fileID, dirDeviceID = toInteger devID}
-
+  ResolvePath path -> join <$> ((identifyPath path <$> Posix.getFileStatus path) `catchingIO` FileReadError path)
```

Comment:
Does `Posix.getFileStatus` work on Windows? [Their docs](https://hackage.haskell.org/package/unix-2.7.2.2) imply the answer is no:
> The package is not supported under Windows.

If not, I think it's not urgent to fix since this isn't used anywhere, but we should have a plan for how we can support windows (or at least throw a nice error message)

Reply:
This is actually from `unix-compat`.

https://github.com/fossas/fossa-cli/pull/862/files#diff-c84a9339a8d0b78e16abd42d8d1933c4138aa85b36da9102625415e726df075dR103

[docs](https://hackage.haskell.org/package/unix-compat-0.5.4/docs/System-PosixCompat-Files.html)

Reply:
whoops 😅 carry on!